src/Extensions/Link/LinkRenderer.php

```diff
@@ -79,7 +79,11 @@ public function setConfiguration(ConfigurationInterface $configuration)
 
     private function isInternalLink(string $url): bool
     {
-        if (str_starts_with($url, config('app.url'))) {
+        $_url         = preg_match('/^https?:\/{1,2}/', $url) ? $url : 'https://'.$url;
```

Comment:
this will only add `https` to the url? so if we would have `http://msq.test` as app url then using `msq.test` would be considered external? can you add a test to this (and maybe a comment) to indicate if this is the case?

Reply:
This line checks if `$url` starts with `http` or `https`, if not it prepends `https` so `parse_url` doesn't return null.

In the next lines I use `parse_url` to be sure to extract only the host part of the string. I apply this technique on both `$url` and `config('app.url')`.

I've already added tests for those cases and you can find them here:
- https://github.com/ArkEcosystem/laravel-commonmark/pull/17/files#diff-2e1984141387d0b12d8630f8e70c666e424fdebfe0106bfeca1e934714ce9eaeR23
- https://github.com/ArkEcosystem/laravel-commonmark/pull/17/files#diff-2e1984141387d0b12d8630f8e70c666e424fdebfe0106bfeca1e934714ce9eaeR41-R42

Reply:
Can't we use a simple `str_starts_with` instead of a regex here if the check is only about the beginning of the string?

Reply:
ah I see, you don't care about the prefix anymore after parse_url is called 👍